src/navigation/breadcrumbs/breadcrumbs.jsx

```diff
@@ -0,0 +1,295 @@
+            '& .navigation_breadcrumbs--breadcrumb_title > p': {
+                fontWeight: fontWeightMedium,
+                fontSize: 12,
+                color: textColorSecondary,
+            },
+            '& .icon-use-path': {
+                transition: `fill ${transitionDurationShortest}ms !important`,
+            },
+            '&:hover': {
+                '& .icon-use-path': {
+                    fill: colorHighlight,
+                },
+                '& .navigation_breadcrumbs--breadcrumb_title > p': {
+                    color: colorHighlight,
+                },
+            },
+        },
+        breadcrumbLast: {
+            '& .navigation_breadcrumbs--breadcrumb_title > p': {
+                fontWeight: fontWeightBold,
+                fontSize: 16,
+                color: textColorPrimary,
+            },
+            '&:hover': {
+                '& .navigation_breadcrumbs--breadcrumb_title > p': {
+                    color: textColorPrimary,
+                },
+            }
+        },
+        breadCrumbSeparator: {
+            display: 'inline',
+            padding: '0 5px 0 0',
+            '& .icon': {
+                marginRight: 5,
+            },
+        },
+        breadCrumbSeparatorTypography: {
+            display: 'inline',
+        },
+        breadcrumbTitle: {
+            display: 'inline',
+            transition: `color ${transitionDurationShortest}ms`,
+        },
+        breadcrumbTitleTypography: {
+            display: 'inline',
+        },
+    };
+});
+
+const BREADCRUMB_TITLE_MAX_LENGTH = 20;
+const ICON_TYPE_FIRST_BREADCRUMB = 'chevron-left';
+
+/**
+ * Indicates the current location inside the application and the previous routes to navigate back.
+ * The breadcrumbs will be automatically generated using the passed react-router core instance (Each route must have a 'title' prop set). 
+ */
+function Breadcrumbs(props) {
+    const {
+        router,
+        showOnlyPreviousRoute,
+        dividerString,
+    } = props;
+
+    const classes = useStyles(props);
+    const [existentRoutes, setExistentRoutes] = useState([]);
+    const [pathBreadcrumbs, setPathBreadcrumbs] = useState([]);
+    const pathName = get(router, 'location.pathname');
+    const pathParams = get(router, 'params');
+    const prevPathName = useRef();
+    const routerPushFunction = get(router, 'push');
+
+    useEffect(() => {
+        const routes = get(router, 'routes', []);
+        const currentExistentRoutes = routesToArray(routes);
+        setExistentRoutes(currentExistentRoutes);
+        const currentBreadCrumbs = getPathNameBreadcrumbs(
+            pathName,
+            pathParams,
+            routerPushFunction,
+            currentExistentRoutes,
+        );
+        setPathBreadcrumbs(currentBreadCrumbs);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []); // didMount
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I'm still a bit nervous that we seem to be running into this issue and having to suppress it.  Could we be doing something not quite right, or not fully understanding something about hooks?

If the code is working, all good I suppose.  Just trying to ask the right questions to justify my time spent code reviewing! =)

Reply:
The code is working, the thing is that the rule is too precise, honestly I can't imagine how could we emulate a `componentDidMount` with a different approach, this was taken out directly from the react docs

![image](https://user-images.githubusercontent.com/8116485/92789510-8adf5400-f370-11ea-87e5-57b9ec13e732.png)

According to the docs, the ESLint rule is a suggestion.

Reply:
@War777 Is that in reference to the empty array as the dependencies, or the suppression of the `react-hooks/exhaustive-deps` ESLINT rule?  Or both?  Please forgive my ignorance here.

Reply:
References the empty array as the dependencies.

Reply:
Yeah, makes sense.  Do you know much about the `react-hooks/exhaustive-deps` ESLINT rule and why we have to suppress it?  Like what is it warning us about and do we need to be concerned?  I tried doing a quick internet search a prior time I encountered this during a code review and only spent a few minutes and don't feel like I understand it well.

Just to be clear ... **_Not a Blocker_** ... I'm just trying to know more.